examples/plot_clusters3.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+"""
+=============================
+Discovering clusters
+=============================
+
+To make use of HDBSCAN as the clustering algorithm used to discver clusters
+you must specify it as the cluster argument. If you wish to specify HDBSCAN
+parameters you will need the dictionary form which includes both the model
+and the params. Since HDBSCAN does not require the number of clusters this
+value to can set to 0. Note that n_clusters much be set to ensure that
+clustering is performed.
```

Comment:
Can you clarify why n_clusters must be set to 0? Should we just automatically set it to 0 if HDBSCAN clustering is used (instead of making the user pass it as an arg)

Reply:
Looking through the code it was required than ``n_clusters`` be set to something other than ``None`` to trigger the clustering code within ``plot``. If you have a preferred approach I would be happy to change it -- I was trying to make the minimum amount of code changes.

Reply:
Ah thanks, that's right. Hmm, this is a design decision that needs to be made. If `n_clusters` is set to something other than `None`, this triggers `hyp.plot` to run the `cluster` function with the default clustering alg (KMeans). We have at least 2 options here:

1) Change the default of the `cluster` arg to `None` and within `hyp.plot`, and trigger clustering when `cluster` is not None (instead of when `n_clusters` is not `None`).  We could set `n_clusters` to 0 by default. Then, users would have to explicitly pass both the clustering alg and the number of clusters: `hyp.plot(data, cluster='KMeans', n_clusters=3)`

2) Leave the default of `cluster` as KMeans, and continue to trigger clustering when the user changes `n_clusters` to something other than `None`. Make the user pass `n_clusters=0` for `HDBSCAN` to work correctly: `hyp.plot(data, cluster='HDBSCAN', n_clusters=0)`

Some other design? Deal with HDBSCAN as a special case? @lmcinnes does it ever make sense to set `n_clusters`>0, when using HDBSCAN or no?

cc: @jeremymanning